### DISCUSSION TITLE
Hide route from navbar?

### DISCUSSION THREAD

**Discussion by vexilar** [mui/toolpad#4258] [Discussions]
I'm wanting to have routes for /customer/[:id] for create/edit to be able to router.push to them and have them show up in the breadcrumbs in pagelayout, but NOT have them show up in the navigation bar.  Is there a way to achieve this?

**Comment by Janpot** [comment]
Note that you can supply your own title and breadcrumbs to the `PageContainer`

```tsx
<PageContainer
  title="ACME ltd."
  breadCrumbs={[
    { path: '/customer', title: 'Customers' },
    { path: '/customer/12345', title: '12345' },
  ]}
>
  Page content
</PageContainer>
```

This way you can use it on pages that are not matched by any navigation from the `AppProvider`.

---

_We're renaming the `breadCrumbs` prop to `breadcrumbs` in 0.8.0._

**Reply by vexilar** [reply]
Yeah that's what I did as a workaround.  I'm getting this error in the console tho:

`app-index.js:33 Warning: React does not recognize the `breadCrumbs` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `breadcrumbs` instead. If you accidentally passed it from a parent component, remove it from the DOM element.`

**Reply by Janpot** [reply]
That'll also be fixed as part of 0.8.0

**Reply by vexilar** [reply]
Nice.  Thanks for your help

**Reply by Janpot** [reply]
Ok, just to note that we didn't really see this as a workaround, but rather intended usage. It's a simple and versatile API. Always happy to hear ideas about alternatives.

**Reply by aress31** [reply]
A more effective approach would be to include a `hidden` property in the navigation structure, as shown below:

```js
{
  segment: "projects",
  icon: <Folder />,
  title: "Projects",
  // children: [
  //   { segment: `${projectId}/foo` },
  //   { segment: `${projectId}/bar` },
  // ],
}
```

This eliminates the need to modify the breadcrumb integration within the `DashboardLayout` component.

Additionally, implementing index routes that are neither routable nor displayed can help maintain the hierarchical structure of the navigation. You could introduce a property like `index` for this purpose.

**Reply by Janpot** [reply]
I think we'd probably like to avoid encoding information in this navigation structure that applies to  only a single one of the (maybe) rendered descendant components (i.e. `DashboardLayout`). I feel like it makes more sense to move that behavior into the DashboardLayout component itself. For example through a `isNavigationItemVisible` callback prop that is called for each item. This would make it similar to how the X tree view [`isItemDisabled`](https://mui.com/x/react-tree-view/rich-tree-view/items/#disabled-items) prop works.
